### PR TITLE
NullConsoleServices: Minimal implementation to open devices

### PR DESCRIPTION
**Pull request by RAOF**
Android has no VTs, but also doesn't need anything but `open()` to access devices. Rework `NullConsoleServices` to do the minimum required to actually open devices.

Future work: Maybe rename to `MinimalConsoleServices`?

Addresses the review comments for PR: #625. Thanks Andreas!

### CODE REVIEW

**Comment by AlanGriffiths** [comment]
Linker say "./tests/mir_test_doubles/mock_drm.cpp:245: error: undefined reference to 'mir_test_framework::add_open_handler(std::function<std::experimental::fundamentals_v1::optional<int> (char const*, int, unsigned int)>)'
collect2: error: ld returned 1 exit status"

**Comment by AlanGriffiths** [comment]
Well, these test failures seem relevant...
```
[ FAILED ] 7 tests, listed below:
[ FAILED ] MinimalConsoleServicesTest.failure_to_claim_drm_master_is_non_fatal
[ FAILED ] MinimalConsoleServicesTest.doesnt_try_to_set_master_on_non_drm_devices
[ FAILED ] MinimalConsoleServicesTest.tries_to_set_drm_master_on_drm_devices
[ FAILED ] MinimalConsoleServicesTest.failure_to_open_device_node_returns_exceptional_future
[ FAILED ] MinimalConsoleServicesTest.failure_to_open_sys_file_results_in_immediate_exception
[ FAILED ] MinimalConsoleServicesTest.opens_input_devices_in_nonblocking_mode
[ FAILED ] MinimalConsoleServicesTest.does_not_open_drm_devices_in_nonblocking_mode
```

**Comment by RAOF** [comment]
Have I just missed adding a file to git or something‽

**Comment by RAOF** [comment]
*Of course* this doesn't fail in an LXD spread run. That would be too easy…

File: tests/unit-tests/console/test_minimal_console_services.cpp [hunk 1]
```diff
@@ -81,7 +81,7 @@ class MinimalConsoleServicesTest : public testing::Test
                 {
                     if (device_path == path)
                     {
-                        return stub_fd;
+                        return static_cast<int>(stub_fd);
```
Comment (AlanGriffiths) [inline tests/unit-tests/console/test_minimal_console_services.cpp]: Ugly, but essentially preexisting (as there's no Fd::as_int())

**Pull request merged**